app/assets/javascripts/discourse/app/services/key-value-store.js

```diff
@@ -0,0 +1,32 @@
+import Service from "@ember/service";
+import KeyValueStore from "discourse/lib/key-value-store";
+
+const PROXIED_METHODS = [
+  "abandonLocal",
+  "remove",
+  "set",
+  "setObject",
+  "get",
+  "getInt",
+  "getObject",
+  "getItem",
+  "removeItem",
+  "setItem",
+];
+
+/**
+ * This is the global key-value-store which is injectable as a service.
+ * Alternatively, consumers can use `discourse/lib/key-value-store` directly
+ * to create their own namespaced store.
+ * */
+export default class KeyValueStoreService extends Service {
+  init() {
+    super.init(...arguments);
+
+    this._keyValueStore = new KeyValueStore("discourse_");
+
```

Comment:
Can this be a constructor instead (since it's a native class)?

```suggestion
  _keyValueStore = new KeyValueStore("discourse_");

  constructor() {
    super(...arguments);
```